lib/ably/rest/client.rb

```diff
@@ -461,7 +466,40 @@ def auth_request_timeout
         end
       end
 
+      # If the primary host endpoint fails, and a subsequent fallback host succeeds, the fallback
+      #   host that succeeded is used for +ClientOption+ +fallback_retry_timeout+ seconds to avoid
+      #   retries to known failing hosts for a short period of time.
+      # See https://github.com/ably/docs/pull/554, spec id #RSC15f
+      #
+      # @return [nil, String]  Returns nil (falsey) if the primary host is being used, or the currently used host if a fallback host is currently preferred
+      def using_preferred_fallback_host?
+        if preferred_fallback_connection && (preferred_fallback_connection.fetch(:expires_at) > Time.now)
+          preferred_fallback_connection.fetch(:connection_object).url_prefix.to_s.gsub(%r{^\w+://}, '').gsub(%r{:\d+/?$}, '')
```

Comment:
I don't think that was available in the object when I checked, because it doesn't actually know the host, it defines a url_suffix oddly when you create a connection object.

Reply:
Ok, not sure what the hell I was doing, thanks for spotting and forcing me to check again :)